pkg/datastore/mysql/model_wrapper.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+func wrapModel(entity interface{}) (interface{}, error) {
+	switch e := entity.(type) {
+	case *model.Project:
+		if e == nil {
+			return nil, fmt.Errorf("nil entity given")
+		}
+		return &project{
+			Project: *e,
+			Extra:   e.Id,
+		}, nil
+	case *model.Application:
+		if e == nil {
+			return nil, fmt.Errorf("nil entity given")
+		}
+		return &application{
+			Application: *e,
+			Extra:       e.Name,
+		}, nil
+	case *model.Command:
+		if e == nil {
+			return nil, fmt.Errorf("nil entity given")
+		}
+		return &command{
+			Command: *e,
+			Extra:   e.Id,
+		}, nil
+	case *model.Deployment:
+		if e == nil {
+			return nil, fmt.Errorf("nil entity given")
+		}
+		return &deployment{
+			Deployment: *e,
+			Extra:      e.ApplicationName,
```

Comment:
Honestly I don't get exactly why Extra exists. Tell me a little bit about why ApplicationName is used 😄

Reply:
👍  @khanhtc1202 Can you comment about that to this file and link to the RFC.
> Another option is to create a stored generated column that shadows value from an extra attribute of JSON data, then we could create indexes (FULLTEXT or normal) on that column for search features.

Reply:
Of course, it can be done by another PR if you want.

Reply:
This `_extra` field is just for future required queries such as query with `LIKE` operator. As I noted on [RFC-0003](https://github.com/pipe-cd/pipe/blob/master/docs/rfcs/0003-sql-datastore.md#text-search-operations-on-specific-json-field), it looks like indexes on virtual generated columns are not hit in `LIKE` query, which could lead to the performance decrease issue. To avoid that, we add an `_extra` attribute to our JSON data model, and shadow it in as table column by 
```
extra VARCHAR(100) GENERATED ALWAYS AS (data->>"$.extra") STORED
```
(it also helps us avoid using ALTER TABLE if we want to update this extra column since it always shadows value from JSON data)
That was the story 🙏 
In case of the deployment model, there is no deployment name so instead of id (random UUID), I think using the deployment's application name would be more useful.

Reply:
@khanhtc1202  I meant adding the comment to the source code, at the top of this file.
It would help other people later.

Reply:
Sure, will address it on this PR 🙏

Reply:
Addressed on https://github.com/pipe-cd/pipe/pull/1653/commits/2b3aef01453487c78a6f7e960e78c20f8f759f4e 🙏

Reply:
That makes sense. Thank you for letting me know!